Pirate-Wires/pirate-wires
# Add Email Signup Modal to Article Pages

### Notion Spec
- https://www.notion.so/Add-email-signup-modal-to-all-article-pages-502ce8ab96c34c1aab3ef143f09f6a7f?pvs=4

### Demo

https://github.com/Pirate-Wires/pirate-wires/assets/2637355/e28b2121-a7dd-4d7e-8558-d8f5a0dbf3c8


https://github.com/Pirate-Wires/pirate-wires/assets/2637355/77d5e6a3-84d2-4faf-a656-339596d605b6


### Testing Instructions
** Ensure that `SITE_URL` is set to `http://localhost:3000` in your `.env.local` (not `https://localhost:3000`)

1. Run the app in a desktop browser as a signed out user (use an incognito tab and clear localstorage)
2. Navigate to any article
3. Scroll down ~35% of the vertical length of the article
4. Verify that the newsletter sign up modal appears (floating up from the bottom), the background overlay fades in, and the button in the lower righthand corner showing the number of free articles remaining appears
5. Verify that as you scroll further, the modal is locked in the vertical center of the viewport
6. Click the `X` icon on the "Free articles remaining" button
7. Verify that the "Free articles remaining" button disappears 
8. Click `Continue` 
9. Verify that the modal and background overlay disappear
10. Reload the page or navigate to a different article 
11. Scroll to the bottom of the article 
12. Verify that the modal UI does not appear
13. Clear `localstorage`
14. Repeat steps 1-3
15. When the modal appears, enter your email and click "Sign In"
16. Complete the sign in flow
17. Verify that you are redirected to the article you were initially reading
18. Scroll to the bottom of the article 
19. Verify that the modal UI does not appear
20. Sign out
21. Navigate to any article
22. add the query parameter `_cio_id=123` to the url string and reload the page
23. Scroll to the bottom of the article 
24. Verify that the modal UI does not appear
25. Load a new incognito browser session
26. Implement the following diff:
```
diff --git a/app/(website)/p/[slug]/page.tsx b/app/(website)/p/[slug]/page.tsx
index 8af9efa..897b0f2 100644
--- a/app/(website)/p/[slug]/page.tsx
+++ b/app/(website)/p/[slug]/page.tsx
@@ -87,7 +87,7 @@ export default async function PostDefault({ params }) {

   if (!session && post.slug) {
     const header = headers();
-    const ipAddress = (header.get("x-forwarded-for") ?? "127.0.0.1").split(",")[0];
+    const ipAddress = "127.0.0.456";

     const viewedArticles = await getViewedArticles(ipAddress);
     freeViewedArticleCount = viewedArticles.length;
```
27. Navigate to any article
28. Scroll down far enough for the modal UI to appear
29. Verify that the text "2 free articles remaining" appears on the button in the lower right corner
30. Without dismissing the modal, navigate to a new article
31. Repeat step 28
32. Verify that the text "1 free article remaining" appears on the button in the lower right corner
33. Repeat 30
34. Verify that instead of the email signup modal, the `Continue Reading With a Free Trial` bottom bar appears
35. Load a new incognito browser session
36. Navigate to an individual article
37. Subscribe to the article
38. Navigate to another article from the same newsletter
39. Verify that the modal UI does not appear
40. Navigate to an article in a different newsletter
41. Verify that the modal UI appears
